spring-graphql/src/main/java/org/springframework/graphql/data/method/annotation/support/DataFetcherHandlerMethod.java

```diff
@@ -113,7 +114,13 @@ public Object invoke(DataFetchingEnvironment environment) {
 		}
 
 		if (Arrays.stream(args).noneMatch(arg -> arg instanceof Mono)) {
-			return validateAndInvoke(args, environment);
+			Object result = validateAndInvoke(args, environment);
+
+			if (result instanceof CompletableFuture<?>) {
+				return Mono.fromFuture((CompletableFuture<?>) result);
+			}
+
+			return result;
```

Comment:
Not sure if more types should be checked here? Or whether this "shortcut" of not processing the return value should be removed completely, because even when there aren't any `Mono` types in the arguments, the return type still could be a `Publisher` or `CompletableFuture`.